### PR TITLE
In page edit

### PR DESCRIPTION
**Pull request by akucharska**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

### CODE REVIEW

File: src/components/App/ExtensibilityRoutes.js [hunk 1]
```diff
@@ -35,50 +44,84 @@ const ColumnWrapper = ({ defaultColumn = 'list', resourceType }) => {
...
-              customResourceName={layoutState?.midColumn?.resourceName}
-              customNamespaceId={layoutState.midColumn?.namespaceId}
-            />
-          </div>
-        )
+      layout={
+        !midColumnComponent ? 'OneColumn' : layoutState?.layout || 'OneColumn'
       }
+      startColumn={<div slot="">{startColumnComponent}</div>}
+      midColumn={<div slot="">{midColumnComponent}</div>}
```
Comment (mrCherry97) [inline src/components/App/ExtensibilityRoutes.js:119]: slots should be empty string or `slot="midColumn"` and `slot="startColumn"` ?

File: src/components/App/ExtensibilityRoutes.js [hunk 2]
```diff
@@ -97,7 +140,7 @@ export const createExtensibilityRoutes = (cr, language) => {
         exact
         element={
           <Suspense fallback={<Spinner />}>
-            <ColumnWrapper resourceType={urlPath} />
+            <ColumnWrapper resourceType={urlPath} cr={cr} />
```
Comment (mrCherry97) [inline src/components/App/ExtensibilityRoutes.js:143]: `cr` is not so clear to understand what this prop is passing, please rename this

File: src/components/Clusters/components/KubeconfigUpload/KubeconfigUpload.js [hunk 3]
```diff
@@ -52,10 +52,10 @@ export function KubeconfigUpload({ kubeconfig, setKubeconfig, formRef }) {
         modeSelectorDisabled={
           kubeconfig ? !Object.keys(kubeconfig)?.length : !!!kubeconfig
         }
-        noAdvancedMode={true}
         initialMode={'MODE_YAML'}
         className="kubeconfig-upload__form add-cluster__content-container"
-        yamlSearchHidden={true}
+        yamlSearchDisabled={true}
+        yamlHideDisabled={true}
```
Comment (mrCherry97) [inline src/components/Clusters/components/KubeconfigUpload/KubeconfigUpload.js:58]:
```suggestion
        yamlSearchDisabled
        yamlHideDisabled
```

File: src/resources/CustomResourceDefinitions/CRCreate.js [hunk 4]
```diff
@@ -12,10 +13,15 @@ function CRCreate({
   onChange,
   formElementRef,
   crd,
-  toggleFormFn,
   layoutNumber,
+  resource: initialCustomResource,
+  ...props
```
Comment (mrCherry97) [inline src/resources/CustomResourceDefinitions/CRCreate.js:18]: If I see well, another props are not used anywhere here.

File: src/resources/createResourceRoutes.js [hunk 5]
```diff
@@ -47,100 +56,147 @@ export const createKubernetesUrl = ({
...
       }
     : null;
 
   useEffect(() => {
-    if (layout && props.resourceName && props.resourceType) {
+    if (layout && resourceName && props.resourceType) {
       setLayoutColumn(initialLayoutState);
     }
-  }, [layout, props.resourceName, props.resourceType, props.namespaceId]); // eslint-disable-line react-hooks/exhaustive-deps
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (mrCherry97) [inline src/resources/createResourceRoutes.js:104]:
```suggestion
```

File: src/resources/createResourceRoutes.js [hunk 6]
```diff
@@ -47,100 +56,147 @@ export const createKubernetesUrl = ({
...
+  }, [
+    layout,
+    isColumnLeyoutEnabled,
+    namespaceId,
+    resourceName,
+    props.resourceType,
+  ]);
+
+  const layoutCloseCreateUrl = resourceListUrl({
+    kind: props.resourceType,
```
Comment (mrCherry97) [inline src/resources/createResourceRoutes.js:114]: Maybe that will be good to have "normal" prop `resourceType` and `resourceName` to not pass it hidden in `props`
```suggestion
    kind: resourceType,
```

File: src/resources/createResourceRoutes.js [hunk 7]
```diff
@@ -47,100 +56,147 @@ export const createKubernetesUrl = ({
...
+  }
+
+  return (
+    <FlexibleColumnLayout
+      style={{ height: '100%' }}
+      layout={
+        !midColumnComponent ? 'OneColumn' : layoutState?.layout || 'OneColumn'
+      }
+      startColumn={<div slot="">{startColumnComponent}</div>}
+      midColumn={<div slot="">{midColumnComponent}</div>}
```
Comment (mrCherry97) [inline src/resources/createResourceRoutes.js:191]: also slots should be empty string?

File: src/resources/other/BusolaExtensions.routes.js [hunk 8]
```diff
@@ -38,32 +50,68 @@ const ColumnWrapper = ({ details }) => {
     if (layout) {
       setLayoutColumn(initialLayoutState);
     }
-  }, [layout, namespace, name]); // eslint-disable-line react-hooks/exhaustive-deps
+  }, [layout, isColumnLeyoutEnabled, namespace, name]); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (mrCherry97) [inline src/resources/other/BusolaExtensions.routes.js:53]:
```suggestion
  }, [layout, isColumnLeyoutEnabled, namespace, name]);
```

File: src/resources/other/CustomResourcesByGroup.routes.js [hunk 9]
```diff
@@ -19,42 +27,180 @@ const CustomResource = React.lazy(() =>
...
+              }
+            : null,
+      }
+    : null;
+
+  useEffect(() => {
+    if (layout) {
+      setLayoutColumn(initialLayoutState);
+    }
+  }, [layout, isColumnLeyoutEnabled, crdName, crName, namespace]); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (mrCherry97) [inline src/resources/other/CustomResourcesByGroup.routes.js:66]:
```suggestion
  }, [layout, isColumnLeyoutEnabled, crdName, crName, namespace]);
```

File: src/resources/other/helmReleases.routes.js [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+        },
+        endColumn: null,
+      }
+    : null;
+
+  useEffect(() => {
+    if (layout) {
+      setLayoutColumn(initialLayoutState);
+    }
+  }, [layout, isColumnLeyoutEnabled, namespace, releaseName]); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (mrCherry97) [inline src/resources/other/helmReleases.routes.js:40]:
```suggestion
  }, [layout, isColumnLeyoutEnabled, namespace, releaseName]);

File: src/shared/ResourceForm/components/CollapsibleSection.js [hunk 11]
```diff
@@ -54,7 +55,9 @@ export function CollapsibleSection({
       <header
         onClick={toggle}
         aria-label={`expand ${title}`}
-        style={{ marginLeft: `${nestingLevel * 16}px` }}
+        style={{
+          marginLeft: `calc(${nestingLevel} * ${spacing.sapUiSmallMarginBegin.marginLeft})`,
+        }}
```
Comment (mrCherry97) [inline src/shared/ResourceForm/components/CollapsibleSection.js:60]: wow, is it really necessary?

**Review by mrCherry97 — CHANGES_REQUESTED** [review]
Question is, when we want to enable column layout, because I'm not big fan of passing `enableColumnLayout` and checking columnLayout feature flag in this bunch of files.

File: src/resources/other/BusolaExtensions.routes.js (hunk 8 above)
Comment (akucharska) [reply src/resources/other/BusolaExtensions.routes.js:53]: Nope, we have `initialLayoutState` and `setLayoutColumn` there

File: src/resources/createResourceRoutes.js (hunk 5 above)
Comment (akucharska) [reply src/resources/createResourceRoutes.js:104]: Nope, we have `initialLayoutState` and `setLayoutColumn` there

File: src/resources/other/CustomResourcesByGroup.routes.js (hunk 9 above)
Comment (akucharska) [reply src/resources/other/CustomResourcesByGroup.routes.js:66]: We can not remove it

File: src/resources/other/helmReleases.routes.js (hunk 10 above)
Comment (akucharska) [reply src/resources/other/helmReleases.routes.js:40]: We can not remove it

File: src/shared/ResourceForm/components/CollapsibleSection.js (hunk 11 above)
Comment (akucharska) [reply src/shared/ResourceForm/components/CollapsibleSection.js:60]: We probably will change it in the next task but setting proper margin for nested form sections was wild

**Pull request merged**